pvlib/test/test_pvsystem.py

```diff
@@ -353,37 +353,235 @@ def test_PVSystem_calcparams_desoto(cec_module_params):
+    },
+    {  # Can handle all rank-1 singleton array inputs
+     'Rsh': np.array([20.]),
+     'Rs': np.array([0.1]),
+     'nNsVth': np.array([0.5]),
+     'I': np.array([3.]),
+     'I0': np.array([6.e-7]),
+     'IL': np.array([7.]),
+     'V_expected': np.array([7.5049875193450521])
+    },
+    {  # Can handle all rank-1 non-singleton array inputs with infinite shunt
+      #  resistance, Rsh=inf gives V=Voc=nNsVth*(np.log(IL + I0) - np.log(I0)
+      #  at I=0
+      'Rsh': np.array([np.inf, 20.]),
+      'Rs': np.array([0.1, 0.1]),
+      'nNsVth': np.array([0.5, 0.5]),
+      'I': np.array([0., 3.]),
+      'I0': np.array([6.e-7, 6.e-7]),
+      'IL': np.array([7., 7.]),
+      'V_expected': np.array([0.5*(np.log(7. + 6.e-7) - np.log(6.e-7)),
+                             7.5049875193450521])
+    },
+    {  # Can handle mixed inputs with a rank-2 array with infinite shunt
+      #  resistance, Rsh=inf gives V=Voc=nNsVth*(np.log(IL + I0) - np.log(I0)
+      #  at I=0
+      'Rsh': np.array([[np.inf, np.inf], [np.inf, np.inf]]),
+      'Rs': np.array([0.1]),
+      'nNsVth': np.array(0.5),
+      'I': 0.,
+      'I0': np.array([6.e-7]),
+      'IL': np.array([7.]),
+      'V_expected': 0.5*(np.log(7. + 6.e-7) - np.log(6.e-7))*np.ones((2, 2))
+    },
+    {  # Can handle ideal series and shunt, Rsh=inf and Rs=0 give
+      #  V = nNsVth*(np.log(IL - I + I0) - np.log(I0))
+      'Rsh': np.inf,
+      'Rs': 0.,
+      'nNsVth': 0.5,
+      'I': np.array([7., 7./2., 0.]),
+      'I0': 6.e-7,
+      'IL': 7.,
+      'V_expected': np.array([0., 0.5*(np.log(7. - 7./2. + 6.e-7) -
+                              np.log(6.e-7)), 0.5*(np.log(7. + 6.e-7) -
+                              np.log(6.e-7))])
+    },
+    {  # Can handle only ideal series resistance, no closed form solution
+      'Rsh': 20.,
+      'Rs': 0.,
+      'nNsVth': 0.5,
+      'I': 3.,
+      'I0': 6.e-7,
+      'IL': 7.,
+      'V_expected': 7.804987519345062
+    },
+    {  # Can handle all python scalar inputs with big LambertW arg
+      'Rsh': 500.,
+      'Rs': 10.,
+      'nNsVth': 4.06,
+      'I': 0.,
+      'I0': 6.e-10,
+      'IL': 1.2,
+      'V_expected': 86.320000493521079
+    },
+    {  # Can handle all python scalar inputs with bigger LambertW arg
+      #  1000 W/m^2 on a Canadian Solar 220M with 20 C ambient temp
+      #  github issue 225 (this appears to be from PR 226 not issue 225)
+      'Rsh': 190.,
+      'Rs': 1.065,
+      'nNsVth': 2.89,
+      'I': 0.,
+      'I0': 7.05196029e-08,
+      'IL': 10.491262,
+      'V_expected': 54.303958833791455
+    },
+    {  # Can handle all python scalar inputs with bigger LambertW arg
+      #  1000 W/m^2 on a Canadian Solar 220M with 20 C ambient temp
+      #  github issue 225
+      'Rsh': 381.68,
+      'Rs': 1.065,
+      'nNsVth': 2.681527737715915,
+      'I': 0.,
+      'I0': 1.8739027472625636e-09,
+      'IL': 5.1366949999999996,
+      'V_expected': 58.19323124611128
```

Comment:
@cwhanse Would you mind running this parameter set though the Matlab version to "independently" verify my V_expected value?

Reply:
58.193231246111300

Reply:
Awesome! Thanks.